connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/CorrelationResult.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound;
+
+public record CorrelationResult(CorrelationResultCode code) {
+
+  public enum CorrelationResultCode {
+    OK(true),
+    ACTIVATION_CONDITION_NOT_MET(false);
+
+    private final boolean isCorrelated;
+
+    public boolean isCorrelated() {
+      return isCorrelated;
+    }
+
+    CorrelationResultCode(boolean isCorrelated) {
+      this.isCorrelated = isCorrelated;
+    }
+  }
```

Comment:
What about a sealed `CorrelationResult` class and `CorrelationResult.Success` and `CorrelationResult.Failed(reason)` subclasses?

Reply:
Would this also mean returning e.g. Zeebe connection errors in the response? (in other words would this method then never throw exceptions?)

Reply:
Yes, that was the idea. As I think it would be cumbersome to deal with the result object in addition to exceptions. WDYT?

Reply:
Agreed, done